File: backend/GameApp.Application/UseCases/EnemyUseCases/EnemyGetUseCase.cs


using GameApp.Domain.Entities;
using GameApp.Domain.ValueObjects.Enemies;

namespace GameApp.Application.UseCases.EnemyUseCases;

public interface EnemyGetUseCase
{
    public Task<Enemy?> GetEnemy(Guid id);

    public Task<Enemy?> GetEnemyByName(EnemyName name);

    public Task<IEnumerable<Enemy>> GetAllEnemys();
}

File: frontend/gameapp-frontend/src/application/usecases/enemy-use-cases/enemy-get-use-case.ts
import type { Enemy } from "../../../domain/entities/enemy";
import type { EnemyName } from "../../../domain/value-objects/enemies/enemy-name";

export interface IEnemyGetUseCase {
    getEnemy(id: string): Promise<Enemy>;
    getEnemyByName(enemyName: EnemyName): Promise<Enemy>;
    getAllEnemys(): Promise<Enemy[]>;
}
